src/definition/messages/IMessage.ts

```diff
@@ -25,4 +25,8 @@ export interface IMessage {
     parseUrls?: boolean;
     customFields?: { [key: string]: any };
     blocks?: Array<IBlock>;
+    starred?: Array<{ _id: string }>;
+    pinned?: boolean;
+    pinnedAt?: Date;
+    pinnedBy?: { _id: string, username: string };
```

Comment:
I would request that this be typed instead of generic.

Reply:
I am not so sure about this, as changing it to `IUser` would require us to add _yet another_ roundtrip to the DB to convert the message data to the app's format. This would be added to every event we listen to, and we're adding listeners to even more events. That's why I worry about the perf impact

Reply:
I am saying something such as `IUserLookup` which would have the value that is defined here.

```ts
export interface IUserLookup {
  _id: string;
  username: string;
}
```

Reply:
Oh I see!